src/renderer/dx/DxFontRenderData.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "../../renderer/inc/FontInfoDesired.hpp"
+#include "BoxDrawingEffect.h"
+
+#include <dwrite.h>
+#include <dwrite_1.h>
+#include <dwrite_2.h>
+#include <dwrite_3.h>
+
+#include <wrl.h>
+
+namespace Microsoft::Console::Render
+{
+    class DxFontRenderData
+    {
+    public:
+        struct LineMetrics
+        {
+            float gridlineWidth;
+            float underlineOffset;
+            float underlineOffset2;
+            float underlineWidth;
+            float strikethroughOffset;
+            float strikethroughWidth;
+        };
+
+        DxFontRenderData(::Microsoft::WRL::ComPtr<IDWriteFactory1> dwriteFactory) noexcept;
+
+        // DirectWrite text analyzer from the factory
+        [[nodiscard]] Microsoft::WRL::ComPtr<IDWriteTextAnalyzer1> Analyzer() noexcept;
+
+        [[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFallback> GetSystemFontFallback();
+
+        [[nodiscard]] til::size GlyphCell() noexcept;
+        [[nodiscard]] LineMetrics GetLineMetrics() noexcept;
+
+        // The DirectWrite format object representing the size and other text properties to be applied (by default)
+        [[nodiscard]] Microsoft::WRL::ComPtr<IDWriteTextFormat> DefaultTextFormat() noexcept;

```

Comment:
Originally I called this `RegularTextFormat`. Then I realize that users could set this a bold variant of the font like `Hack Bold`. Then it would confuse people what the `Regular` means. So I changed this to `DefaultTextFormat`. Any better idea?

Reply:
Default is fine for now. I thought maybe "normal" but that also implies something that might not be true.